src/libs/ck-libs/ampi/ampi.C

```diff
@@ -2140,70 +2143,28 @@ ampi::~ampi() noexcept
 //------------------------ Communicator Splitting ---------------------
 class ampiSplitKey {
  public:
-  int nextSplitComm;
+  int nextComm; //MPI_Comm value of next comm
   int color; //New class of processes we'll belong to
   int key; //To determine rank in new ordering
   int rank; //Rank in old ordering
+  int type; //Type of comm: intra, inter, cart, graph, etc.
   ampiSplitKey() noexcept {}
-  ampiSplitKey(int nextSplitComm_,int color_,int key_,int rank_) noexcept
-    :nextSplitComm(nextSplitComm_), color(color_), key(key_), rank(rank_) {}
+  ampiSplitKey(int nextComm_,int color_,int key_,int rank_,AmpiCommType type_) noexcept
+    :nextComm(nextComm_), color(color_), key(key_), rank(rank_), type((int)type_) {}
 };
 
-#define MPI_INTER 10
-
 /* "type" may indicate whether call is for a cartesian topology etc. */
-void ampi::split(int color,int key,MPI_Comm *dest, int type) noexcept
+void ampi::split(int color,int key,MPI_Comm *dest, AmpiCommType type) noexcept
 {
-  if (type == MPI_CART) {
-    ampiSplitKey splitKey(parent->getNextCart(),color,key,myRank);
-    int rootIdx=myComm.getIndexForRank(0);
-    CkCallback cb(CkIndex_ampi::splitPhase1(0),CkArrayIndex1D(rootIdx),myComm.getProxy());
-    contribute(sizeof(splitKey),&splitKey,CkReduction::concat,cb);
-
-    ampi * dis = block(); //Resumed by ampi::registrationFinish
-    MPI_Comm newComm = dis->parent->getNextCart()-1;
-    *dest=newComm;
-  }
-  else if (type == MPI_GRAPH) {
-    ampiSplitKey splitKey(parent->getNextGraph(),color,key,myRank);
-    int rootIdx=myComm.getIndexForRank(0);
-    CkCallback cb(CkIndex_ampi::splitPhase1(0),CkArrayIndex1D(rootIdx),myComm.getProxy());
-    contribute(sizeof(splitKey),&splitKey,CkReduction::concat,cb);
-
-    ampi * dis = block(); //Resumed by ampi::registrationFinish
-    MPI_Comm newComm = dis->parent->getNextGraph()-1;
-    *dest=newComm;
-  }
-  else if (type == MPI_DIST_GRAPH) {
-    ampiSplitKey splitKey(parent->getNextDistGraph(),color,key,myRank);
-    int rootIdx=myComm.getIndexForRank(0);
-    CkCallback cb(CkIndex_ampi::splitPhase1(0),CkArrayIndex1D(rootIdx),myComm.getProxy());
-    contribute(sizeof(splitKey),&splitKey,CkReduction::concat,cb);
-
-    ampi * dis = block(); //Resumed by ampi::registrationFinish
-    MPI_Comm newComm = dis->parent->getNextDistGraph()-1;
-    *dest=newComm;
-  }
-  else if (type == MPI_INTER) {
-    ampiSplitKey splitKey(parent->getNextInter(),color,key,myRank);
-    int rootIdx=myComm.getIndexForRank(0);
-    CkCallback cb(CkIndex_ampi::splitPhaseInter(0),CkArrayIndex1D(rootIdx),myComm.getProxy());
-    contribute(sizeof(splitKey),&splitKey,CkReduction::concat,cb);
+  MPI_Comm nextComm = parent->getNextComm();
+  ampiSplitKey splitKey(nextComm,color,key,myRank,type);
+  int rootIdx=myComm->getIndexForRank(0);
+  CkCallback cb(CkIndex_ampi::splitPhase1(0),CkArrayIndex1D(rootIdx),myComm->getProxy());
```

Comment:
Does there still need to be a check here for MPI_INTER to call splitPhaseInter instead of splitPhase1?

Reply:
Yes, fixed